### PR TITLE
feat(di): introduce di and use it for services

### LINKED ISSUE #364: dependency management of singelton services

### PR DESCRIPTION
**Pull request by sk91**
resolves #364

### CODE REVIEW

**Comment by sk91** [comment]
solution inspired by [https://medium.com/@Jeffijoe/dependency-injection-in-node-js-2016-edition-f2a88efdd427](this post)

using the [awilix](https://github.com/jeffijoe/awilix) di container

File: src/main/main.js [hunk 1]
```diff
@@ -99,30 +93,22 @@ function onReady(app) {
 			appUpdater();
 		}
 		await db.init();
-		if (!isDevMode() && !isTestMode()) {
-			await crashReportService.startCrashReport();
-		}
 		const store = configureStore(global.state, 'main');
+		const container = configureContrainer(store, app);
+		const di = container.cradle;
```
Comment (rodrigopavezi) [inline src/main/main.js]: @sk91 would make more sense to call this `applicationContext` instead of di? Thoughts?

Comment (rodrigopavezi) [reply src/main/main.js]: Also, there is a typo on method name and I think it should be `configureContainer`. Or if we are calling applicationContext we could call it `configureContext`. Thoughts?

**Comment by rodrigopavezi** [comment]
looks good 👍

**Pull request merged**